### PR TITLE
[CAS-1363] Add possibility to add scroll listener to the MessageListView

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-1363

### 🎯 Goal

Add  possibility to add a custom scroll listener to the inner list of the `MessageListView`.

### 🛠 Implementation details

Added methods to register and unregister scroll listeners. Also, exposing the inner `RecyclerView` so that clients are able to use it, for example, to calculate scrolling offset.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF

giphy

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt [hunk 1]
```diff
@@ -545,6 +546,35 @@ public class MessageListView : ConstraintLayout {
...
+    }
+
+    /**
+     * Returns the inner [RecyclerView] that is used to display a list of message list items.
+     *
+     * @return The inner [RecyclerView] with messages.
+     */
+    public fun getInnerMessageList(): RecyclerView {
+        return binding.chatMessagesRV
+    }
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt:576]: What's the use case for exposing this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -61,6 +61,7 @@
 ### ⬆️ Improved
 
 ### ✅ Added
+- Added `MessageListView::addOnScrollListener` and `MessageListView::removeOnScrollListener` methods to the `MessageListView` to listen to scroll events of the inner message list.
```
Comment (zsmb13) [inline CHANGELOG.md:64]: Shouldn't we also add an example of how to use these APIs to the docs?

Comment (bychkovdmitry) [reply CHANGELOG.md:64]: Seems like a request from a customer for a functionality which will be not very popular among other clients. What do you think, should we create an example anyway?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt:555]: Decided to expose just the inner RecyclerView.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt [hunk 3]
```diff
@@ -545,6 +545,15 @@ public class MessageListView : ConstraintLayout {
         super.onDetachedFromWindow()
     }
 
+    /**
+     * Returns the inner [RecyclerView] that is used to display a list of message list items.
+     *
+     * @return The inner [RecyclerView] with messages.
+     */
+    public fun getInnerMessageList(): RecyclerView {
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt:553]: Perhaps `getRecyclerView` for the method name? As a customer, I don't know what I'd expect to be the "inner message list".

Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListView.kt:553]: 👍

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**